cardano-node-emulator/src/Cardano/Node/Emulator/Effects.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE KindSignatures    #-}
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module Cardano.Node.Emulator.Effects(
+  -- * Node client
+    NodeClientEffect(..)
+  , publishTx
+  , getClientSlot
+  , getClientParams
+  ) where
+
+import Cardano.Node.Emulator.Params (Params)
+import Control.Monad.Freer.TH (makeEffect)
+import Ledger (CardanoTx, Slot)
+
+data NodeClientEffect r where
```

Comment:
I'm beginning to think that this should stay in `plutus-contract`...

This effect seems to be for supporting multiple node backends in the emulator. So by interpreting this effect differently, you can use a real node, or the API provided by `cardano-node-emulator`.

In the future, if `cardano-node-emulator` uses the same IPC interface as `cardano-node`, then this effect would not be necessary anymore.

What do you think?